apps/browser/supabase/migrations/20231206012608_chat_history.sql

```diff
@@ -0,0 +1,83 @@
+drop table roles CASCADE;
+drop table chats CASCADE;
+drop table messages CASCADE;
+
+CREATE TABLE "public"."chats" (
+  "id" "uuid" DEFAULT "extensions"."uuid_generate_v4"() NOT NULL,
+  "created_at" timestamp with time zone DEFAULT "now"() NOT NULL,
+  "user_id" "uuid" default auth.uid(),
+  PRIMARY KEY ("id"),
+  FOREIGN KEY ("user_id") REFERENCES "public"."users"("id")
+);
+
+alter table "public"."chats" enable row level security;
+
+create table "public"."logs" (
+  "id" "uuid" DEFAULT "extensions"."uuid_generate_v4"() NOT NULL,
+  "created_at" timestamp with time zone not null default now(),
+  "title" text not null,
+  "content" text,
+  "chat_id" uuid not null,
+  "user" text not null,
+  FOREIGN KEY ("chat_id") REFERENCES "public"."chats"("id")
+);
+
+alter table "public"."logs" enable row level security;
+
+create table "public"."variables" (
+  "id" "uuid" DEFAULT "extensions"."uuid_generate_v4"() NOT NULL,
+  "key" text not null,
+  "value" text not null,
+  "chat_id" uuid not null,
+  FOREIGN KEY ("chat_id") REFERENCES "public"."chats"("id")
+);
+
+alter table "public"."variables" enable row level security;
+
+CREATE TABLE "public"."messages" (
+  "id" "uuid" DEFAULT "extensions"."uuid_generate_v4"() NOT NULL,
+  "created_at" timestamp with time zone DEFAULT "now"() NOT NULL,
+  "chat_id" uuid NOT NULL,
+  "content" "text",
+  "name" "text",
+  "function_call" "json",
+  "temporary" boolean NOT NULL,
+  "tool_call_id" "text",
+  "tool_calls" json,
+  "role" "text" NOT NULL,
+  PRIMARY KEY ("id"),
```

Comment:
Why do we not add `PRIMARY KEY ("id")` for the other tables we're creating (logs, variables)?

Reply:
by creating the primary key using a unique index it helps with large datasets when it comes to query things. since there will be a lot more logs & variables i thought it would be better to add this performance improvement for the long term.

with your comment i noticed that i missed this for messages, so i will update it accordingly

Reply:
update: i've added the `PRIMARY KEY` to `logs` and `variables` and just added the index using chat_id for these three tables (`logs`, `variables` and `messages`)